search/api.py

```diff
@@ -90,83 +90,106 @@ def _index_users_chunk(users):
     conn = get_conn()
     insert_count, errors = bulk(
         conn,
-        (serialize_user(user) for user in users),
+        (serialize_program_enrolled_user(program_enrollment) for program_enrollment in program_enrollments),
         index=settings.ELASTICSEARCH_INDEX,
         doc_type=USER_DOC_TYPE,
     )
-
     if len(errors) > 0:
         raise ReindexException("Error during bulk insert: {errors}".format(
             errors=errors
         ))
     refresh_index()
-
     return insert_count
 
 
-def index_users(users, chunk_size=100):
+def index_program_enrolled_users(program_enrollments, chunk_size=100):
     """
-    Add/update profile records in Elasticsearch.
+    Add/update ProgramEnrollment records in Elasticsearch.
 
     Args:
-        users (iterable of User):
-            Iterable of users
+        program_enrollments (iterable of ProgramEnrollments):
+            Iterable of ProgramEnrollments to serialize and index
         chunk_size (int):
             How many users to index at once.
 
     Returns:
         int: Number of indexed items
     """
     # Use an iterator so we can keep track of what's been indexed already
-    users = iter(users)
-
+    program_enrollments = iter(program_enrollments)
     count = 0
-    chunk = list(islice(users, chunk_size))
+    chunk = list(islice(program_enrollments, chunk_size))
     while len(chunk) > 0:
-        count += _index_users_chunk(chunk)
-        chunk = list(islice(users, chunk_size))
-
+        count += _index_program_enrolled_users_chunk(chunk)
+        chunk = list(islice(program_enrollments, chunk_size))
     refresh_index()
-
     return count
 
 
-def remove_user(user):
+def index_users(users, chunk_size=100):
     """
-    Remove a user from Elasticsearch.
+    Indexes a list of users via their ProgramEnrollments
+    """
+    program_enrollments = ProgramEnrollment.objects.filter(user__in=users).select_related('user', 'program').all()
+    return index_program_enrolled_users(program_enrollments, chunk_size)
+
+
+def remove_program_enrolled_user(program_enrollment):
+    """
+    Remove a program-enrolled user from Elasticsearch.
     """
     conn = get_conn()
     try:
-        conn.delete(index=settings.ELASTICSEARCH_INDEX, doc_type=USER_DOC_TYPE, id=user.id)
+        conn.delete(index=settings.ELASTICSEARCH_INDEX, doc_type=USER_DOC_TYPE, id=program_enrollment.id)
     except NotFoundError:
         # Item is already gone
         pass
 
 
-def serialize_user(user):
+def remove_user(user):
+    """
+    Remove a user from Elasticsearch.
+    """
+    program_enrollments = ProgramEnrollment.objects.filter(user=user).select_related('user', 'program').all()
+    for program_enrollment in program_enrollments:
+        remove_program_enrolled_user(program_enrollment)
```

Comment:
From the code coverage report it looks like this line is not triggered by tests. Can you add or adjust a test to cover this line?